metrics/web/docs/features/deep_links/01_deep_linking.md

```diff
@@ -142,3 +171,232 @@ history.replaceState(null, 'metrics', '$uri');
+
+So, the main idea of this approach is to create a bidirectional connection between the `NavigationNotifier` and the `Page ChangeNotifiers`, which allows to:
+- Provide the page parameters updates to `Page ChangeNotifiers`, which apply them (e.g., apply filtering criteria, searching values, etc.);
+- Provide the page parameters updates to `NavigationNotifier` (e.g., when the user applies a filter, searches, etc.), which updates the URL and the browser history.
+
+Consider the following component diagram that briefly describes this approach:
+
+![Navigation notifier approach diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design/metrics/web/docs/features/deep_links/diagrams/navigation_notifier_approach_component_diagram.puml)
+
+Let's sum up the pros and cons of this approach:
+
+Pros:
+- Well segregated responsibility between components;
+- Extensible approach - meaning that we can easily add new page-specific deep links or extend them;
+- Well-testable because of the responsibility segregation.
+
+Cons:
+- Requires extending the existing `ChangeNotifier`s.
+
+#### Route Parameters approach
+The main idea of this approach is to pass the deep links directly to the specific pages that will handle them. When the page parameters change (e.g., in response to sorting, filtering, etc.), the pages use the `NavigationNotifier` to update the URL and the query parameters without changing the browser history.
+
+Consider the next component diagram that illustrates this approach:
+
+![Route parameters approach diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/Flank/flank-dashboard/deep_links_design/metrics/web/docs/features/deep_links/diagrams/route_parameters_approach_component_diagram.puml)
+
+Let's review the pros and cons of the `route parameters` approach:
+
+Pros:
+- Does not require implementing new or extending existing `ChangeNotifier`s.
+
+Cons:
+- Does not satisfy the existing architecture requirements (see this [section](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md#ui-elements) of the ["Metrics Web Presentation Layer"](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md) document); 
+- Less testable;
+- High coupling;
+- Adding parameters handling may increase the page widgets size, and decrease the code readability.
+
+### Program
+> Detailed solution description to class/method level.
+
+Comparing the approaches described above, we choose the [`NavigationNotifier`](#navigationnotifier-approach) approach as it aligns with the overall architectural requirements, simplifies the testing process, and is highly extensible.
```

Comment:
:readable: Maybe it would be better to move this sentence to the `Architecture` section?